vadl/test/vadl/lcb/riscv/riscv32/GenerateLinkerComponentsRiscv32PassTest.java

```diff
@@ -0,0 +1,263 @@
+        new PassKey(GenerateLinkerComponentsPass.class.getName()));
+    var passManager = setup.passManager();
+
+    var generatedLinkerComponents =
+        (GenerateLinkerComponentsPass.Output) passManager.getPassResults()
+            .lastResultOf(GenerateLinkerComponentsPass.class);
+    var generatedVariantKinds = generatedLinkerComponents.variantKinds();
+
+    expectedVariantKinds().forEach(
+        expected -> Assertions.assertTrue(
+            generatedVariantKinds.stream()
+                .anyMatch(generated -> generated.value().equals(expected.value())
+                    && generated.human().equals(expected.human())
+                    && generated.isImmediate() == expected.isImmediate()
+                ),
+            "Expected variant kind: " + expected + " not found in generated variant kinds: "
+                + generatedVariantKinds
+        )
+    );
+  }
+
+  private static Stream<VariantKind> expectedVariantKinds() {
+    return Stream.of(
+        VariantKind.none(),
+        VariantKind.invalid(),
+        new VariantKind("VK_ABS_RV3264I_hi", "hi", false),
+        new VariantKind("VK_ABS_RV3264I_lo", "lo", false),
+        new VariantKind("VK_SYMB_ABS_RV3264I_Itype_imm", "SYMB_ABS_RV3264I_Itype_imm", true),
+        new VariantKind("VK_SYMB_PCREL_RV3264I_Itype_imm", "SYMB_PCREL_RV3264I_Itype_imm", true),
+        new VariantKind("VK_SYMB_ABS_RV3264I_Utype_imm", "SYMB_ABS_RV3264I_Utype_imm", true),
+        new VariantKind("VK_SYMB_PCREL_RV3264I_Utype_imm", "SYMB_PCREL_RV3264I_Utype_imm", true),
+        new VariantKind("VK_SYMB_ABS_RV3264I_Stype_imm", "SYMB_ABS_RV3264I_Stype_imm", true),
+        new VariantKind("VK_SYMB_PCREL_RV3264I_Stype_imm", "SYMB_PCREL_RV3264I_Stype_imm", true),
+        new VariantKind("VK_SYMB_ABS_RV3264I_Btype_imm", "SYMB_ABS_RV3264I_Btype_imm", true),
+        new VariantKind("VK_SYMB_PCREL_RV3264I_Btype_imm", "SYMB_PCREL_RV3264I_Btype_imm", true),
+        new VariantKind("VK_SYMB_ABS_RV3264I_Jtype_imm", "SYMB_ABS_RV3264I_Jtype_imm", true),
+        new VariantKind("VK_SYMB_PCREL_RV3264I_Jtype_imm", "SYMB_PCREL_RV3264I_Jtype_imm", true),
+        new VariantKind("VK_SYMB_ABS_RV3264I_Ftype_sft", "SYMB_ABS_RV3264I_Ftype_sft", true),
+        new VariantKind("VK_SYMB_PCREL_RV3264I_Ftype_sft", "SYMB_PCREL_RV3264I_Ftype_sft", true),
+        new VariantKind("VK_DECODE_RV3264I_Rtype_shamt", "DECODE_RV3264I_Rtype_shamt", true),
+        new VariantKind("VK_DECODE_RV3264I_Itype_immS", "DECODE_RV3264I_Itype_immS", true),
+        new VariantKind("VK_DECODE_RV3264I_Utype_immUp", "DECODE_RV3264I_Utype_immUp", true),
+        new VariantKind("VK_DECODE_RV3264I_Stype_immS", "DECODE_RV3264I_Stype_immS", true),
+        new VariantKind("VK_DECODE_RV3264I_Btype_immS", "DECODE_RV3264I_Btype_immS", true),
+        new VariantKind("VK_DECODE_RV3264I_Jtype_immS", "DECODE_RV3264I_Jtype_immS", true),
+        new VariantKind("VK_DECODE_RV3264I_Ftype_shamt", "DECODE_RV3264I_Ftype_shamt", true)
```

Comment:
minor: please also add the tests for `pcrel_hi`, `pcrel_lo` and `got_hi`

`to32AndHi`, `to32AndLo`, `lowerHalfHi` and `lowerHalfLo` can be skipped